Dependencies/napi/napi-direct/include/napi/napi-inl.h

```diff
@@ -1781,6 +1781,24 @@ inline Value Function::Call(napi_value recv, const std::vector<napi_value>& args
   return Call(recv, args.size(), args.data());
 }
 
+inline Value Function::Call(napi_value recv, size_t argc, const Napi::Value* args) const {
+    napi_value stackArgs[6];
+    std::vector<napi_value> heapArgs;
+    napi_value* argv;
+    if (argc <= std::size(stackArgs)) {
+        argv = stackArgs;
+    } else {
+        heapArgs.resize(argc);
+        argv = heapArgs.data();
+    }
+
+    for (size_t i = 0; i < argc; ++i) {
+        argv[i] = args[i];
+    }
+
+    return Call(recv, argc, argv);
+}
+
```

Comment:
This file is mostly unmodified from the node repo. If this is actually necessary, then we should try to upstream it to node. We typically don't change the N-API contract.

Reply:
Filed a PR here: https://github.com/nodejs/node-addon-api/pull/1026